### PR TITLE
[numpy] add torch.argwhere

### PR DESCRIPTION
**Pull request by kshitij12345**
Adds `torch.argwhere` as an alias to `torch.nonzero`

Currently, `torch.nonzero` is actually provides equivalent functionality to `np.argwhere`.

From NumPy docs, 
> np.argwhere(a) is almost the same as np.transpose(np.nonzero(a)), but produces a result of the correct shape for a 0D array.

### CODE REVIEW

**Comment by kshitij12345** [comment]
Gentle ping @mruberry @saketh-are

File: torch/_torch_docs.py
Comment (saketh-are) [inline torch/_torch_docs.py:5949]: I found the "grouped by element" language from numpy's documentation to be pretty informal and hard to understand. Looks like I'm also not the first one: https://stackoverflow.com/questions/52400354/how-to-understand-the-np-argwhere-function/52400485.

It seems that "grouped by element" is stated in numpy.argwhere's documentation in reference to numpy.nonzero's behavior (https://numpy.org/doc/stable/reference/generated/numpy.nonzero.html), which groups by dimension instead.

For our purposes it seems clearer to reuse/refer to the language from torch.nonzero's documentation of its as_tuple=False case: https://pytorch.org/docs/stable/generated/torch.nonzero.html.

**Review by saketh-are — APPROVED** [review]
Thanks for implementing this @kshitij12345. I left a minor comment regarding the documentation of argwhere's output format.

File: torch/_torch_docs.py
Comment (kshitij12345) [reply torch/_torch_docs.py:5949]: That makes sense. Thanks!

**Comment by kshitij12345** [comment]
@saketh-are should be ready for another review. Thanks :)

File: torch/_torch_docs.py
Comment (saketh-are) [inline torch/_torch_docs.py:5875]: `torch.nonzero` documentation has a warning about host-device synchronization; we should mention it here, too.

File: torch/_torch_docs.py
Comment (saketh-are) [inline torch/_torch_docs.py:5879]: While I recognize that this documentation is copied from `torch.nonzero`'s, this might be a good opportunity to clean up the wording a little. @mruberry rightly pointed out that these two paragraphs seem to be saying the same thing over and over again and could be simplified.

Alternatively, you could file an issue about simplifying both `torch.nonzero` and `torch.argwhere` docs for later followup.

File: torch/testing/_internal/common_methods_invocations.py
Comment (saketh-are) [inline torch/testing/_internal/common_methods_invocations.py:5632]: Could we construct something suitable by hand here instead of indirectly through this randomized routine?

**Review by saketh-are — COMMENTED** [review]
Hey @kshitij12345, thanks for making those changes! I wanted to confirm one last thing with you before we move on to landing this. Given that this PR relies on `torch.nonzeros`'s correctness, would you be able to review the completeness of `torch.nonzero`'s forward tests and make a small comment here about them? I also left a few minor comments inline.

**Comment by kshitij12345** [comment]
>  would you be able to review the completeness of torch.nonzero's forward tests and make a small comment here about them?

Sure. Will have a look :)

**Comment by kshitij12345** [comment]
`torch.nonzero` with `as_tuple=False` is same as `np.argwhere` (this is why we are adding `torch.argwhere` as an alias to `nonzero` with `as_tuple=False`)

`torch.nonzero` with `as_tuple=True` is same as `np.nonzero` (which returns tuple of arrays which can be used for indexing).

<details>

<summary>Implementation</summary>

`nonzero_numpy` corresponds to `as_tuple=True` and `nonzero_cpu` corresponds to `as_tuple=False`

Kernels:
https://github.com/pytorch/pytorch/blob/6a990535154116960767ad422e724250ecfb8325/aten/src/ATen/native/TensorAdvancedIndexing.cpp#L1727-L1746

Dispatch logic:
https://github.com/pytorch/pytorch/blob/6a990535154116960767ad422e724250ecfb8325/torch/csrc/autograd/python_torch_functions_manual.cpp#L673-L689

</details>

There are tests for `nonzero` in test_shape_ops and test_unary_ufuncs. Maybe we can move all of them to one place (and also check for overlap). The main test which checks the `as_tuple=True` is in `test_shape_ops`.

<details>

<summary>test_shape_ops</summary>

https://github.com/pytorch/pytorch/blob/3324bae5f106f22ff78eb6943b79cd607a1b37e4/test/test_shape_ops.py#L568-L616

</details>

<details>

<summary>test_unary_ufuncs</summary>

https://github.com/pytorch/pytorch/blob/3324bae5f106f22ff78eb6943b79cd607a1b37e4/test/test_unary_ufuncs.py#L1441-L1497

</details>

File: torch/testing/_internal/common_methods_invocations.py
Comment (kshitij12345) [reply torch/testing/_internal/common_methods_invocations.py]: Done.

File: torch/_torch_docs.py
Comment (kshitij12345) [reply torch/_torch_docs.py:5872]: Ah. Thanks! Sorry missed that one!

File: torch/_torch_docs.py
Comment (kshitij12345) [reply torch/_torch_docs.py:5876]: I think I will opt filing the issue (assuming we want to/will update `torch.nonzero` to match only the `np.nonzero`)

Thanks!

**Comment by kshitij12345** [comment]
Gentle ping @saketh-are @mruberry

**Review by mruberry — APPROVED** [review]
This looks really good @kshitij12345, I just made a few small comments inline for your review. I'll let @saketh-are merge this when it's ready.

Looking forward, we should consider deprecating nonzero in favor of argwhere. nonzero is popular so this will likely count as a "difficult deprecation," and we should prepare an RFC including it and other difficult deprecations (like max and min) soon to gauge the community's interest in NumPy consistency vs. disruptions.

**Comment by kshitij12345** [comment]
@saketh-are gentle ping :)

**Pull request closed without merging**
